migration/migration2.go

```diff
@@ -0,0 +1,189 @@
+package migration
```

Comment:
This migration is for the changes done in _ffs.FilStorage_:
- Removed _ProposalCid_.
- Removed _ActivationEpoch_.
- Added _DealID_.

If was for the first two bullet points, we wouldn't require a migration since we can leave that data in the datastore and the unmarshaling would just ignore the data. The cause of needed a migration is adding _DealID_. If we don't do something about that, all _DealID_ of existing `ffs.FilStorage` will be 0, which is wrong.

To do this migration we basically leverage the information we have in `/deals/record/<storage-pending/storage-final>/<iid>/<proposalCid>`, so we can map _ProposalCid_ to _DealID_ to fill that attribute.

Reply:
I'll be interested to see how this works in production. For a period of time, when deals started taking over 24 hrs to become active, the code that was updating the deal records would time out and give up on creating a deal record. That has since been fixed, but there may be some deals that don't have corresponding deal records for that reason.